### DISCUSSION TITLE
Next OpenAlea release 2025

### DISCUSSION THREAD

**Discussion by thomasarsouze** [openalea/openalea#258] [Development]
Discussion space related to the next release (spring 2023)  of OpenAlea packages.

**Comment by thomasarsouze** [comment]
Following the meeting of feb 6th 2023, we target a new release of openalea for FSPM meeting 2023, late march 2023:
- include visualea
- Identify the packages to be released
- target python 3.9 to 3.12
- unify namespace
- all packages to be released in openalea3 repo in anaconda
- make sure all packages have tests
- unify minimum GH-actions for all packages with a call to an action in a repo
- update openalea.rtfd.io matrix of released packages with a minimal jupyter notebook 
- if possible make sure to have appropriate documentation of the package

@christian34 @pradal @baugetfa @fredboudon (feel free to add other users)

**Comment by pradal** [comment]
Release 

- [x] core
- [x] grapheditor
- [x] oalab
- [x] stdlib
- [x] visualea

**Reply by pradal** [reply]
@pellem : Retrieve all these packages plus
https://github.com/openalea/openalea-components 

Build a local repo : mamba create -n dev -c openalea3 -c conda-forge openalea.deploy

Clone each repository and install in dev : python setup.py develop

**Reply by pellem** [reply]
Stdlib is in [openalea/components](https://github.com/openalea/openalea-components/tree/master)

**Reply by pellem** [reply]
**1. [Core](https://github.com/openalea/core)**
 - [x] [Setup.py](https://github.com/openalea/core/blob/master/setup.py) : OK
 - [x] [Tests written](https://github.com/openalea/core/tree/master/test) : OK
 - [x] Tests passed : **NOK**

**2. [Grapheditor](https://github.com/openalea/grapheditor)**
- [x]  [Setup.py](https://github.com/openalea/grapheditor/blob/master/setup.py) : OK
- [ ] [Tests written](https://github.com/openalea/grapheditor/tree/master/test) : **NOK**
- [ ] Tests passed : **NOK**

**3. [Oalab](https://github.com/openalea/oalab)**
- [x] [Setup.py](https://github.com/openalea/oalab/blob/master/setup.py) : OK
- [x] [Tests written](https://github.com/openalea/oalab/tree/master/test) : OK
- [ ] Tests passed : **NOK**

**4. [Stdlib](https://github.com/openalea/openalea-components/tree/master/stdlib)** 
- [x] [Setup.py](https://github.com/openalea/openalea-components/blob/master/stdlib/setup.py) : OK
- [x] [Tests written](https://github.com/openalea/openalea-components/tree/master/stdlib/test) : OK
- [x] Tests passed : OK

**5. [Visualea](https://github.com/openalea/visualea)**
- [x] [Setup.py](https://github.com/openalea/visualea/blob/master/setup.py) : OK
- [x] [Tests written](https://github.com/openalea/visualea/tree/master/test) : OK but not exhaustive
- [ ] Tests passed : **NOK**

**Comment by pradal** [comment]
# OpenAlea core (low priority) 
- [ ] openalea-comp / PyLab
- [ ] openalea-comp / NumPy
- [ ] openalea-comp /...

# OpenAlea Packages : 1st layer (high priority)
- [x] mtg
- [x] plantgl
- [x] lpy
- [ ] StructureAnalysis (large work to do)

# OpenAlea Packages : 2nd layer
- [x] hydroroot
- [x] hydroshoot
- [x] Phenomenal
- [x] strawberry
- [x] RSML

# OpenAlea Packages : Work to do on packaging 
- [ ] caribu 
- [ ] PyRATP
- [?] Adel (refactoring)
- [x] EpyMix
- [ ] WheatFSPM

**Comment by pradal** [comment]
Define what is needed for releasing a package

- [x] Documentation
- [x] Tutorial Notebook
- [x] Tutorial VisuAlea
- [x] Tests
- [x] other?

**Comment by pradal** [comment]
@baugetfa : We can start the new release from here

**Reply by baugetfa** [reply]
@pradal: I was wondering if it would not be more clean to start a new discussion because this one is 2 years old. Otherwise let's start it from here.

**Comment by pradal** [comment]
You can of course create a new discussion.
But I have removed the old messages.

I have also updated the project. But we have to manage it differently. 
If you can look at it on how to work with projects it would be nice.

I think I have use the project as a discussion.

Perhaps, an idea: create issues per package with setup, meta, test doc.
And assign it to the release project?

@thomasarsouze ?

**Reply by baugetfa** [reply]
Fair enough, that is fine for me. I'll have a look to the project things. Indeed, we should not make duplication between discussion and project, as for example the pkgs list above and the "conda packaging" in projects.

**Reply by thomasarsouze** [reply]
Agree that we need to learn how to better work with projects, and that we should not define things to do in the Discussion, but rather only reference the project / tickets / issues / todos / etc... here.

Maybe first define a list of packages (done by @pradal I think) and TODO list of things to be done for each package release, and then we can open the issues for each package.

Later, a TODO list for OpenAlea as a whole (rtfd website, gallery website, etc...)

**Comment by pradal** [comment]
Yes. We can copy the list from the project to a discussion.
Then each one can create issues in each package, and then associate the issues to a project.

**Comment by baugetfa** [comment]
It is not so convenient in fact, but we will do with it in fact. For example, every issue that can be followed as to be associated to a given repository. That is logical, but we cannot have a "general issue", a "meta issue", unless keeping the item as draft like this one I modified to set a todo list per pkg [todo pkg](https://github.com/orgs/openalea/projects/9/views/1?pane=issue&itemId=63289482).
Then we can imagine to start for each pkg an issue called release-reviewing that we link to the project, with a status in progress.
At the pull request, the status would change to Review, and when merge to Done.

**Comment by baugetfa** [comment]
# Question:
I started this review processing with hydroroot. I do it on a branch in my fork. But may be it would be more relevant/easy to make this release review from a branch on the Openalea repository ?

**Reply by pradal** [reply]
Yes. When we are few to work on the same package, I recommend to have a share branch where we can both commit

**Reply by pradal** [reply]
And on the project, for each issue, we can associate the issue to the project

**Comment by pradal** [comment]
The release 2025 has been pubblished
